ssl/statem/statem_dtls.c

```diff
@@ -241,7 +241,7 @@ int dtls1_do_write(SSL *s, int type)
 
         ret = dtls1_write_bytes(s, type, &s->init_buf->data[s->init_off], len,
                                 &written);
-        if (ret < 0) {
+        if (ret <= 0) {
```

Comment:
@mattcaswell it is rather odd that dtls1_write_bytes returns 0 when an SSLfatal was called:
https://github.com/openssl/openssl/blob/801c638c50406c93d683c1ab8bd1d430cff4b6d0/ssl/record/rec_layer_d1.c#L825-L828

Reply:
... but there are numerous other places when 0 is returned and written is not initialized either.

Reply:
I think the intention was that `written` is the number of bytes written on success, and you should not count on `written` being updated if you don't get success. However, it might be wise to initialise it to 0 in all cases to avoid mistakes.